### PR TITLE
Fix #1551 - Redirect to the last page when current page is out of range

### LINKED ISSUE #1551: delete item in page n, go back to page 1
information of instance : framabag

if i delete an item in every page n (and n!=1) after i'm being redirect to page 1.
I think i should go back to page n (or n-1 if page n become empty)

### PR DESCRIPTION
**Pull request by delyriand**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documentation | no
| Translation   | no
| Fixed tickets | #1551
| License       | MIT

### CODE REVIEW

File: src/Wallabag/CoreBundle/Controller/EntryController.php [hunk 1]
```diff
@@ -252,7 +253,13 @@ private function showEntries($type, Request $request, $page)
         $entries = new Pagerfanta($pagerAdapter);
 
         $entries->setMaxPerPage($this->getUser()->getConfig()->getItemsPerPage());
-        $entries->setCurrentPage($page);
+        try {
+            $entries->setCurrentPage($page);
+        } catch (OutOfRangeCurrentPageException $e) {
+            if ($page > 1) {
+                return $this->redirect(str_replace('/' . $page, '/' . $entries->getNbPages(), $request->getRequestUri()), 301);
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Controller/EntryController.php]: I'll rather do:

```php
return $this->redirect($this->generateUrl($type, array('page' => $entries->getNbPages())));
```

Comment (delyriand) [inline src/Wallabag/CoreBundle/Controller/EntryController.php]: I agree with you!

File: src/Wallabag/CoreBundle/Controller/EntryController.php [hunk 2]
```diff
@@ -252,7 +253,13 @@ private function showEntries($type, Request $request, $page)
         $entries = new Pagerfanta($pagerAdapter);
 
         $entries->setMaxPerPage($this->getUser()->getConfig()->getItemsPerPage());
-        $entries->setCurrentPage($page);
+        try {
+            $entries->setCurrentPage($page);
+        } catch (OutOfRangeCurrentPageException $e) {
+            if ($page > 1) {
+                return $this->redirect($this->generateUrl($type, array('page' => $entries->getNbPages())), 301);
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Controller/EntryController.php]: Also, could you put a `302` instead? You don't know if in the future you'll have the given page :)

Comment (delyriand) [inline src/Wallabag/CoreBundle/Controller/EntryController.php]: indeed ... in my mind, I wanted to make a temporary redirect :)

**Comment by j0k3r** [comment]
Thanks ! :+1:

**Pull request merged**